File: source/core/sr/activity.cpp
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
// activity.cpp: Implementation definitions for Activity related classes.
//
#include "stdafx.h"
#include "activity.h"

namespace Microsoft {
namespace CognitiveServices {
namespace Speech {
namespace Impl {

nlohmann::json& CSpxActivityJSONAccessor::Get()
{
    return m_accessor();
}

void CSpxActivityJSONAccessor::SetAccessor(accessor_fn accessor)
{
    m_accessor = accessor;
}

nlohmann::json& CSpxActivity::GetJSON()
{
    return m_json;
}

void CSpxActivity::LoadJSON(const nlohmann::json& json)
{
    m_json = json;
}

} } } }

File: source/bindings/java/com/microsoft/cognitiveservices/speech/dialog/Activity.java
//
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
//
package com.microsoft.cognitiveservices.speech.dialog;

import com.microsoft.cognitiveservices.speech.SpeechConfig;
import com.microsoft.cognitiveservices.speech.util.Contracts;

/**
 * Class that defines an Activity.
 */
public class Activity {

    /*! \cond PROTECTED */
    static Class<?> activity = null;


    // load the native library.
    static {
        // trigger loading of native library
        try {
            Class.forName(SpeechConfig.class.getName());
        }
        catch (ClassNotFoundException ex) {
            throw new IllegalStateException(ex);
        }
        activity = Activity.class;
    }

    protected Activity(com.microsoft.cognitiveservices.speech.internal.Activity activity) {
        Contracts.throwIfNull(activity, "activity");
        this.activityImpl = activity;
    }
    /*! \endcond */

    /**
     * Create an activity from a serialized activity.
     * @param activity Serialized activity
     * @return The activity created.
     */
    public static Activity fromSerializedActivity(String activity) {
        Contracts.throwIfNull(activity, "activity");
        return new Activity(com.microsoft.cognitiveservices.speech.internal.Activity.FromSerializedActivity(activity));
    }

    /**
     * Returns the activity serialized as a string.
     * @return The serialized activity.
     */
    public String serialize() {
        return this.activityImpl.Serialize();
    }

    /*! \cond PROTECTED */
    private com.microsoft.cognitiveservices.speech.internal.Activity activityImpl;
    /*! \endcond */

    /**
      * Returns the implementation of the activity.
      * @return The implementation of the activity.
      */
    public com.microsoft.cognitiveservices.speech.internal.Activity getImpl() {
        return this.activityImpl;
    }
}
